### DISCUSSION TITLE
Setting .tsconfig strict = true breaks the example

### DISCUSSION THREAD

**Discussion by nidegen** [vercel/next.js#62490] [Help] [answered]
### Summary

I've been using the `with-cloudinary` example and noticed if I enabled `strict` TS mode, at least VS Code finds a lot of errors.

I am pretty new to Next.js, but I wonder why this is OK. What speaks against strict mode enabled?

https://github.com/vercel/next.js/blob/e98978033fd7401a7e09e376a57afdde0d22c0bd/examples/with-cloudinary/tsconfig.json#L7

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Yeah this should be fixed though. I checked the errors, and there are a couple of things that could be, kind of, problematic.

I'll open a PR with a fix.

Why is it allowed? In the original PR nobody commented on it, but generally, one would allow strict to be false, when one wants to be friendly to beginners. Just like you are raising an eyebrow to strict being false, there are people who are also not happy with the usage of TypeScript at all. 

Since that gets to a matter of preference, I won't emit any judgement, people are at different stages of learning, and everyone is free to choose which path they follow. Learning material and such, can at best try to accommodate for everyone. 

That's my guess though, because I've seen people complain about the examples using TS at all.

**Reply by nidegen** [reply]
Awesome, that clears it up! Let me know when you have the PR ready:)

**Reply by icyJoseph** [reply]
Yeah, I'll try to complete it sometime this week, no promises though. I saw that a library used in the project, `react-use-keypress`, doesn't support types, not even in `definitelyTyped` as `@types/react-use-keypress`, so we'll need to generate a declaration for it, or switch the library...

I start to see why it was kept as strict false 😅
